Applications/Spire/Source/Ui/QuantitySpinBox.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Spire/Ui/QuantitySpinBox.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  auto real_to_quantity(RealSpinBox::Real value) {
+    auto quantity = Quantity::FromValue(value.str(
+      std::numeric_limits<Quantity>::digits10, std::ios_base::dec));
+    if(quantity != boost::none) {
+      return *quantity;
+    }
+    throw std::runtime_error(R"(QuantitySpinBox: failed to convert Real to
+      Quantity.")");
+  }
+}
+
+QuantitySpinBox::QuantitySpinBox(Quantity value, QWidget* parent)
+    : QAbstractSpinBox(parent) {
+  m_spin_box = new RealSpinBox(display_string(value).c_str(), this);
```

Comment:
Every call to ```display_string``` follows up with a call to ```c_str()``` and then appears to construct a ```Real```. Factor all this out into a ```to_real``` function that converts a ```Quantity``` into a ```Real```.